libnetwork/drivers/bridge/network_linux_test.go

```diff
@@ -121,24 +119,17 @@ func TestLinkCreateNoEnableIPv6(t *testing.T) {
 	}
 
 	ipdList := getIPv4Data(t)
-	err := d.CreateNetwork("dummy", option, nil, ipdList, getIPv6Data(t))
-	if err != nil {
-		t.Fatalf("Failed to create bridge: %v", err)
-	}
+	err = d.CreateNetwork("dummy", option, nil, ipdList, getIPv6Data(t))
+	assert.NilError(t, err, "Failed to create bridge")
+
 	te := newTestEndpoint(ipdList[0].Pool, 30)
 	err = d.CreateEndpoint(context.Background(), "dummy", "ep", te.Interface(), nil)
-	if err != nil {
-		t.Fatalf("Failed to create a link: %s", err.Error())
-	}
-
-	iface := te.iface
-	if iface.addrv6 != nil && iface.addrv6.IP.To16() != nil {
-		t.Fatalf("Expected IPv6 address to be nil when IPv6 is not enabled. Got IPv6 = %s", iface.addrv6.String())
-	}
+	assert.NilError(t, err)
 
-	if te.gw6.To16() != nil {
-		t.Fatalf("Expected GatewayIPv6 to be nil when IPv6 is not enabled. Got GatewayIPv6 = %s", te.gw6.String())
+	if !assert.Check(t, is.Nil(te.iface.addrv6), "Expected IPv6 address to be nil when IPv6 is not enabled") {
+		assert.Check(t, is.Nil(te.iface.addrv6.IP.To16()), "Expected IPv6 address to be nil when IPv6 is not enabled. got: %s", te.iface.addrv6.String())
 	}
+	assert.Check(t, is.Nil(te.gw6.To16()), "Expected GatewayIPv6 to be nil when IPv6 is not enabled. Got GatewayIPv6 = %s", te.gw6.String())
```

Comment:
This is doing what it did before, but I'm not sure why the `To16` stuff is needed ... `addrv6` and `gw6` should both just be nil if IPv6 is disabled. So it's fine as it is, or can be simplified to ...

```suggestion
	assert.Check(t, is.Nil(te.iface.addrv6), "Expected IPv6 address to be nil when IPv6 is not enabled, got %s", te.iface.addrv6)
	assert.Check(t, is.Nil(te.gw6), "Expected GatewayIPv6 to be nil when IPv6 is not enabled, got %s", te.gw6)
```

Reply:
Thank you! All of this cleanup is going to help a lot.